src/Lumi/Components2/Text.purs

```diff
@@ -0,0 +1,270 @@
+module Lumi.Components2.Text
+  ( Text
+  , TextProps, TextType
+  , text
+
+  , TextModifier
+  , body, strong, emphasized, subtext
+  , muted, color
+
+  , ParagraphProps
+  , paragraph_, paragraph
+
+  , Header
+  , HeaderProps
+  , subsectionHeader
+  , sectionHeader
+  , title
+  , mainHeader
+  ) where
+
+import Prelude
+
+import Data.Maybe (Maybe(..), fromMaybe, maybe)
+import Effect.Unsafe (unsafePerformEffect)
+import Lumi.Components (LumiComponent, PropsModifier, PropsModifier', lumiComponent, propsModifier)
+import Lumi.Components.Color (ColorMap)
+import Lumi.Styles (Style, StyleModifier, StyleProperty)
+import Lumi.Styles as S
+import Lumi.Styles.Theme (LumiTheme(..), TextMap, textFontSize, textLineHeight, textMargin, useTheme)
+import React.Basic (JSX, ReactComponent)
+import React.Basic.DOM as R
+import React.Basic.Emotion as E
+import React.Basic.Hooks as Hooks
+
+-- Text
+
+data Text = Text
+
+data TextType
```

Comment:
One problem with data constructors is that you can't add a new one without it being a breaking change. Did you consider perhaps only exporting the data type `TextType`, and then value synonyms for each constructor (i.e. `body = Body`)?

Reply:
That's what's being done here already. The constructors of `TextType` are not being exported. Instead we export the modifiers that set the corresponding text type on an element:
```purescript
body :: TextModifier Text
body = propsModifier _{ type = Just Body }

subtext :: TextModifier Text
subtext = propsModifier _{ type = Just Subtext }

...
```

Reply:
Oh, so it is, sorry!

Reply:
No worries :)